cnd/src/swap_protocols/halight/connector.rs

```diff
@@ -436,6 +436,23 @@ impl WaitForCancelled for LndConnectorAsReceiver {
     }
 }
 
+#[cfg(target_os = "macos")]
```

Comment:
Please keep the block under the `cfg` attribute to a minimum to avoid involuntary discrepancies between macos and non macos code:

This compiles:
```
fn client(certificate: &Certificate, macaroon: &Macaroon) -> Result<reqwest::Client, Error> {
    let cert = certificate.0.clone();
    let mut default_headers = HeaderMap::with_capacity(1);
    default_headers.insert(
        "Grpc-Metadata-macaroon",
        HeaderValue::from_str(&macaroon.0)?,
    );

    #[cfg(target_os = "macos")]
    let client = reqwest::Client::builder()
        .danger_accept_invalid_certs(true)
        .add_root_certificate(cert)
        .default_headers(default_headers)
        .build()?;

    #[cfg(not(target_os = "macos"))]
    let client = reqwest::Client::builder()
        .danger_accept_invalid_certs(true)
        .add_root_certificate(cert)
        .default_headers(default_headers)
        .build()?;

    Ok(client)
}
```

Also, as per @thomaseizinger's comment, provide more details and why we are doing that and why the risk is minimal.

Reply:
updated + added comment

Reply:
Actually, we could initialize the boolean value based on the target os :)